zone/zonedb.cpp

```diff
@@ -2346,253 +2347,131 @@ const NPCType *ZoneDatabase::LoadNPCTypesData(uint32 npc_type_id, bool bulk_load
-		where_condition.c_str()
-	);
 
-	auto results = QueryDatabase(query);
-	if (!results.Success()) {
-		return nullptr;
+		filter = fmt::format(
+			SQL(
+				id IN (
+					select npcID from spawnentry where spawngroupID IN (
+						select spawngroupID from spawn2 where `zone` = '{}' and `version` = {}
+					)
+				)
+			),
+			zone->GetShortName(),
+			zone->GetInstanceVersion()
+		);
 	}
 
-	for (auto row = results.begin(); row != results.end(); ++row) {
-		NPCType *temp_npctype_data;
-		temp_npctype_data = new NPCType;
-		memset(temp_npctype_data, 0, sizeof *temp_npctype_data);
-
-		temp_npctype_data->npc_id = atoi(row[0]);
-
-		strn0cpy(temp_npctype_data->name, row[1], 50);
-
-		temp_npctype_data->level              = atoi(row[2]);
-		temp_npctype_data->race               = atoi(row[3]);
-		temp_npctype_data->class_             = atoi(row[4]);
-		temp_npctype_data->max_hp             = atoi(row[5]);
-		temp_npctype_data->current_hp         = temp_npctype_data->max_hp;
-		temp_npctype_data->Mana               = atoi(row[6]);
-		temp_npctype_data->gender             = atoi(row[7]);
-		temp_npctype_data->texture            = atoi(row[8]);
-		temp_npctype_data->helmtexture        = atoi(row[9]);
-		temp_npctype_data->herosforgemodel    = atoul(row[10]);
-		temp_npctype_data->size               = atof(row[11]);
-		temp_npctype_data->loottable_id       = atoi(row[12]);
-		temp_npctype_data->merchanttype       = atoi(row[13]);
-		temp_npctype_data->alt_currency_type  = atoi(row[14]);
-		temp_npctype_data->adventure_template = atoi(row[15]);
-		temp_npctype_data->trap_template      = atoi(row[16]);
-		temp_npctype_data->attack_speed       = atof(row[17]);
-		temp_npctype_data->STR                = atoi(row[18]);
-		temp_npctype_data->STA                = atoi(row[19]);
-		temp_npctype_data->DEX                = atoi(row[20]);
-		temp_npctype_data->AGI                = atoi(row[21]);
-		temp_npctype_data->INT                = atoi(row[22]);
-		temp_npctype_data->WIS                = atoi(row[23]);
-		temp_npctype_data->CHA                = atoi(row[24]);
-		temp_npctype_data->MR                 = atoi(row[25]);
-		temp_npctype_data->CR                 = atoi(row[26]);
-		temp_npctype_data->DR                 = atoi(row[27]);
-		temp_npctype_data->FR                 = atoi(row[28]);
-		temp_npctype_data->PR                 = atoi(row[29]);
-		temp_npctype_data->Corrup             = atoi(row[30]);
-		temp_npctype_data->PhR                = atoi(row[31]);
-		temp_npctype_data->min_dmg            = atoi(row[32]);
-		temp_npctype_data->max_dmg            = atoi(row[33]);
-		temp_npctype_data->attack_count       = atoi(row[34]);
-
-		if (row[35] != nullptr) {
-			strn0cpy(temp_npctype_data->special_abilities, row[35], 512);
+	for (NpcTypesRepository::NpcTypes &n : NpcTypesRepository::GetWhere((Database &) content_db, filter)) {
+		NPCType *t;
+		t = new NPCType;
+		memset(t, 0, sizeof *t);
+
+		t->npc_id = n.id;
+
+		strn0cpy(t->name, n.name.c_str(), 50);
```

Comment:
Not really important since this is unchanged but was there a reason this was hardcoded to 50 before instead of using `sizeof(t->name)`? The name is a `char name[64]`

Reply:
I have no idea to be honest